crates/router/src/services/api.rs

```diff
@@ -410,79 +414,100 @@ pub async fn send_request(
     option_timeout_secs: Option<u64>,
 ) -> CustomResult<reqwest::Response, errors::ApiClientError> {
     logger::debug!(method=?request.method, headers=?request.headers, payload=?request.payload, ?request);
-    let url = &request.url;
+
+    let url = reqwest::Url::parse(&request.url)
+        .into_report()
+        .change_context(errors::ApiClientError::UrlEncodingFailed)?;
+
     #[cfg(feature = "dummy_connector")]
-    let should_bypass_proxy = url.starts_with(&state.conf.connectors.dummyconnector.base_url)
-        || client::proxy_bypass_urls(&state.conf.locker).contains(url);
+    let should_bypass_proxy = url
+        .as_str()
+        .starts_with(&state.conf.connectors.dummyconnector.base_url)
+        || client::proxy_bypass_urls(&state.conf.locker).contains(&url.to_string());
     #[cfg(not(feature = "dummy_connector"))]
-    let should_bypass_proxy = client::proxy_bypass_urls(&state.conf.locker).contains(url);
+    let should_bypass_proxy = client::proxy_bypass_urls(&state.conf.locker).contains(url.as_str());
     let client = client::create_client(
         &state.conf.proxy,
         should_bypass_proxy,
         request.certificate,
         request.certificate_key,
     )?;
     let headers = request.headers.construct_header_map()?;
-    match request.method {
-        Method::Get => client.get(url),
-        Method::Post => {
-            let client = client.post(url);
-            match request.content_type {
-                Some(ContentType::Json) => client.json(&request.payload),
-
-                Some(ContentType::FormData) => client.multipart(
-                    request
-                        .form_data
-                        .unwrap_or_else(reqwest::multipart::Form::new),
-                ),
 
-                // Currently this is not used remove this if not required
-                // If using this then handle the serde_part
-                Some(ContentType::FormUrlEncoded) => {
-                    let payload = match request.payload.clone() {
-                        Some(req) => serde_json::from_str(req.peek())
+    let metrics_tag = router_env::opentelemetry::KeyValue {
+        key: consts::METRICS_HOST_TAG_NAME.into(),
+        value: url.host_str().unwrap_or_default().to_string().into(),
+    };
+
+    let send_request = async {
```

Comment:
If applicable, can we Box Pin this future .

Reply:
Tested with `cargo clippy -- -W clippy::large_futures`. Not applicable